### ISSUE TITLE
use v:this_session for save workspace

### ISSUE THREAD

**Issue by chumager** [vim-ctrlspace/vim-ctrlspace#259]
**Is your feature request related to a problem? Please describe.**
Hi, I make an Issue yesterday related to problems with the cs_workspaces file saved after the last repository update... 
What I didn't realize is when I save the workspace (with the old version who works for me) it just don't save, with the message "Nothing to save".
Because I edit several projects at the same time I've lots of tabs in different paths, and my linters need to be executed in the path the file resides, so I added this to mi .vimrc
```
autocmd BufEnter,BufNewFile * silent! lcd %:p:h
```
I think this change, and because you try to find the file to save the session instead using ```v:this_session``` causes the plugin can't find where to save the workspace.

**Describe the solution you'd like**
As Vim knows where is the session file (v:this_session), I think using this file as start point to solve relative paths will fix all this problems.

**Describe alternatives you've considered**
In general I maintain my vim session always open and use CS to name tabs and to avoid saving session, so I've no other alternatives.

**Version(s) (please complete the following information):**
 - OS: Debian buster 10.2
 - Version: VIM - Vi IMproved 8.1 (2018 May 18, compiled Jun 15 2019 16:41:15)
Parches incluidos: 1-875, 878, 884, 948, 1046, 1365-1368, 1382, 1401
Modificado por team+vim@tracker.debian.org
Compilado por team+vim@tracker.debian.org
Versión "enorme" con interfaz gráfica de GTK2.
  Aspectos incluidos (+) o no (-):
+acl               +extra_search      +mouse_netterm     +tag_old_static
+arabic            +farsi             +mouse_sgr         -tag_any_white
+autocmd           +file_in_path      -mouse_sysmouse    +tcl
+autochdir         +find_in_path      +mouse_urxvt       +termguicolors
-autoservername    +float             +mouse_xterm       +terminal
+balloon_eval      +folding           +multi_byte        +terminfo
+balloon_eval_term -footer            +multi_lang        +termresponse
+browse            +fork()            -mzscheme          +textobjects
++builtin_terms    +gettext           +netbeans_intg     +textprop
+byte_offset       -hangul_input      +num64             +timers
+channel           +iconv             +packages          +title
+cindent           +insert_expand     +path_extra        +toolbar
+clientserver      +job               +perl              +user_commands
+clipboard         +jumplist          +persistent_undo   +vartabs
+cmdline_compl     +keymap            +postscript        +vertsplit
+cmdline_hist      +lambda            +printer           +virtualedit
+cmdline_info      +langmap           +profile           +visual
+comments          +libcall           -python            +visualextra
+conceal           +linebreak         +python3           +viminfo
+cryptv            +lispindent        +quickfix          +vreplace
+cscope            +listcmds          +reltime           +wildignore
+cursorbind        +localmap          +rightleft         +wildmenu
+cursorshape       +lua               +ruby              +windows
+dialog_con_gui    +menu              +scrollbind        +writebackup
+diff              +mksession         +signs             +X11
+digraphs          +modify_fname      +smartindent       -xfontset
+dnd               +mouse             +startuptime       +xim
-ebcdic            +mouseshape        +statusline        +xpm
+emacs_tags        +mouse_dec         -sun_workshop      +xsmp_interact
+eval              +mouse_gpm         +syntax            +xterm_clipboard
+ex_extra          -mouse_jsbterm     +tag_binary        -xterm_save
     archivo "vimrc" del sistema: "$VIM/vimrc"
     archivo "vimrc" del usuario: "$HOME/.vimrc"
  2º archivo "vimrc" del usuario: "~/.vim/vimrc"
      archivo "exrc" del usuario: "$HOME/.exrc"
    archivo "gvimrc" del sistema: "$VIM/gvimrc"
    archivo "gvimrc" del usuario: "$HOME/.gvimrc"
 2º archivo "gvimrc" del usuario: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
     archivo de menú del sistema: "$VIMRUNTIME/menu.vim"
            predefinido para $VIM: "/usr/share/vim"
Compilación: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-2.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/gio-unix-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/libpng16 -Wdate-time  -g -O2 -fdebug-prefix-map=/build/vim-4Pursk/vim-8.1.0875=. -fstack-protector-strong -Wformat -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1     
Enlazado: gcc   -L. -Wl,-z,relro -Wl,-z,now -fstack-protector -rdynamic -Wl,-export-dynamic -Wl,-E  -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -o vim   -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lfontconfig -lfreetype -lSM -lICE -lXpm -lXt -lX11 -lXdmcp -lSM -lICE  -lm -ltinfo -lnsl  -lselinux  -lacl -lattr -lgpm -ldl  -L/usr/lib -llua5.2 -Wl,-E  -fstack-protector-strong -L/usr/local/lib  -L/usr/lib/x86_64-linux-gnu/perl/5.28/CORE -lperl -ldl -lm -lpthread -lcrypt  -L/usr/lib/python3.7/config-3.7m-x86_64-linux-gnu -lpython3.7m -lcrypt -lpthread -ldl -lutil -lm -L/usr/lib/x86_64-linux-gnu -ltcl8.6 -ldl -lz -lpthread -lm -lruby-2.5 -lpthread -lgmp -ldl -lcrypt -lm

**Comment by Konfekt** [comment]
Yes, this autocmd is delicate together with this plug-in as amply documented in

https://github.com/vim-ctrlspace/vim-ctrlspace/issues?utf8=%E2%9C%93&q=autochdir

(Note that `autochdir` is a built-in setting for achieving for your autocmd intends to.)

Could switching to the repo's root dir by an autocmd like

```vim
    autocmd BufRead,BufNewFile,BufFilePost *
            \ if !exists('g:SessionLoad') && !empty(FugitiveGitDir(bufnr('%'))) && empty(&l:buftype) && filereadable(expand('%:p')) |
            \   Glcd |
            \ endif
```

soothe your needs?

**Comment by chumager** [comment]
Hi, AFAIK Glcd is a vim-easygit command... I don't use it...

I think this is a design issue more than a behavior it self... vim provides a way to know the path of my session file, so why bother trying to find the session file if I can simply ask it to vim?

Seems cleaner to me... don't you think?

Regards.

**Comment by Konfekt** [comment]
> Hi, AFAIK Glcd is a vim-easygit command... I don't use it...

It is a command defined in `vim-fugitive`. For an alternative suggestion without any need for a plug-in, see https://github.com/vim-ctrlspace/vim-ctrlspace/pull/44/files

> why bother trying to find the session file if I can simply ask it to vim?

Not sure what exactly is intended, but other plug-ins, such as `vim-startify` offer convenient session handling.

**Comment by Konfekt** [comment]
Please check whether https://github.com/vim-ctrlspace/vim-ctrlspace/pull/268 aleviates these issues.
